x/exercise.rb

```diff
@@ -0,0 +1,41 @@
+module X
+  # Essentially a wrapper around the /exercise resource from the exercises API.
+  class Exercise
+    module Readme
+      def self.find(track_id, slug)
+        Exercise.new('track_id' => track_id, 'slug' => slug).readme
+      end
+    end
+
+    module TestFiles
+      def self.find(track_id, slug)
+        Exercise.new('track_id' => track_id, 'slug' => slug).test_files
+      end
+    end
+
+    METHODS = [
+      :track_id, :slug
+    ]
+    attr_reader(*METHODS)
+
+    def initialize(attrs={})
+      METHODS.each do |name|
+        instance_variable_set(:"@#{name}", attrs[name.to_s])
+      end
+    end
+
+    def test_files
+      data('tests')
+    end
+
+    def readme
+      data('readme')
+    end
+
+    private
+    def data(endpoint)
```

Comment:
I'd like to have the HTTP call happen in a class method, and let the `Exercise` instance be a dumb object (like with `X::Track`).